game-core/src/main/java/games/strategy/triplea/attachments/UnitSupportAttachment.java

```diff
@@ -337,7 +350,7 @@ static void addRule(final UnitType type, final GameData data, final boolean firs
   private static Set<UnitType> getTargets(final GameData data) {
     Set<UnitType> types = null;
     for (final UnitSupportAttachment rule : get(data)) {
-      if (rule.getBonusType().equals(Constants.OLD_ART_RULE_NAME)) {
+      if (rule.getBonusType().getSecond().equals(Constants.OLD_ART_RULE_NAME)) {
```

Comment:
To note potential further benefit of a value object, a value object for bonus type could encapsulate this logic and remove redundancy, eg:
```if(rule.getBonusType().isOldArtilleryRule())```

Reply:
Updated.